### PR TITLE
[WIP] Add utilities

### PR DESCRIPTION
**Pull request by maya**
### ⚠️🚧 work in progress 🚧 ⚠️

This begins work for adding in utilities as well as rearchitecting the Sass to support this.

- Add utilities folder
- Break up mixins into individual files
- Rearrange Sass imports in a more logical order (resembling ITCSS)

Related #2264, #2272, #2214.

[😎 Preview](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/add-utilities)

---
## Notes
- There are two deprecated files: `core/_grid-settings.scss` has been deprecated for a while and `core/_utilities` (just deprecated). Do we need them for any reason?
- `core/mixins/font-smoothing.scss` has a TODO note that the file can be removed once PostCSS has been added and it has been. We can remove this mixin from the source.

### CODE REVIEW

File: src/stylesheets/uswds.scss [hunk 1]
```diff
@@ -39,3 +58,8 @@
 @import 'components/section';
 @import 'components/sidenav';
 @import 'components/skipnav';
+
+// Utilities ------------ //
+// @import 'utilities/color';
+// @import 'utilities/space';
+// @import 'utilities/typography';
```
Comment (maya) [inline src/stylesheets/uswds.scss]: Should this be removed (or added to)?

**Comment by thisisdano** [comment]
My latest work should address the issues mentioned in https://github.com/uswds/uswds/pull/2378

It should 
- [fix the grid](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/add-utilities/components/detail/grid--default.html)
- [update to the correct blue color](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/add-utilities/components/detail/buttons--default.html)
- [update to the correct light gold color](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/add-utilities/components/detail/alerts--warning.html)
- [use normalize as expected in tables and elsewhere](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/add-utilities/components/detail/tables.html)

**Comment by thisisdano** [comment]
I can merge in `develop` to resolve those conflicts.

**Comment by thisisdano** [comment]
@hursey013 In resolving merge conflicts in `package-lock.json` when merging `develop`, I just chose everything in `develop` — assuming that reflected (y)our new work. But, honestly, not 100% sure. Can you check the diff and validate my choices?

**Comment by thisisdano** [comment]
Shoot, it looks like my merge broke something!

**Comment by hursey013** [comment]
@thisisdano I noticed the compiled versions of the stylesheets in `/dist` have a lot of comments in them, and not necessarily in a way that makes sense. `dist/uswds-utilities.css` has 1600+ lines of comments at the top of the file but I'm thinking we are probably safe to just remove them since they make more sense contextually in the Sass files, thoughts?

**Comment by thisisdano** [comment]
Yeah, all that comment stuff is really ugly. Is there a flag we can use to strip those comments from the un-minified css?

File: src/stylesheets/core/_font-definitions.scss [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      300: false,
+      400: 'merriweather-italic-webfont',
+      500: false,
+      600: false,
+      700: false,
+      800: false,
+      900: false,
+    ),
+  ),
+  roboto-mono: (
```
Comment (hursey013) [inline src/stylesheets/core/_font-definitions.scss]: I know we explicitly recommend using Source Sans Pro and Merriweather now - how do Roboto Mono and United Sans work into that recommendation? Are these just to extend the number of available options or are there specific instances when these new font families should be chosen? (Sorry if this has already been covered elsewhere).

Comment (thisisdano) [reply src/stylesheets/core/_font-definitions.scss]: Hasn't been covered yet, but I consider this a topic for upcoming guidance. But also even in the absence of guidance, by starting to build a library of font definitions, we can begin to use the system's new ability to normalize font sizes — and be more simpler and more programmatic about building our `@font-face` rules

**Comment by maya** [comment]
We started with 2 SCSS files in the root and here it's grown to 6. I'm finding this a lot to parse and hard to see which files depend on each other. I think a flat ITCSS structure in a single file could be a lot more linear, simpler, and easier to see. Sticking with a structure that's known and used in the industry could also be easier to maintain. Would this work using the ITCSS structure?

**Comment by thisisdano** [comment]
I think this work is consistent with ITCSS, so maybe we can talk about what would make the organization simpler.

In general, though, there are four files instead of one to make it simpler to include only the parts of the system you want to include:

`uswds-components` is all the components
`uswds-utilities` is the utilities
`uswds-fonts` is the `@font-face` declarations — broken out to make it simpler to inline critical styles in the `head`
`uswds` packages all these together as one

`required.scss` is imports common to all of these — this file could live somewhere else.

I've used it in a project like 
```sass
// project.scss
@import "uswds-utilities"
```
when I didn't want to include the components. You could do a similar thing if you didn't want the utilities. (You can't put `import` directives inside a conditional and control output using, say, an `$output-utilities` bool.)

That said, there's probably another way to organize everything that makes it extra clear. Let's talk it through.

**Comment by joshbruce** [comment]
@Maya and @donjo: Here's an interesting integration we just put on another project I maintain. Having actually put it into practice but I like the idea of it: https://github.com/apps/wip

**Pull request merged**